### PR TITLE
update dev docs to include ROLE ARN setup

### PR DESCRIPTION
**Pull request by nithu0115**
Issue #, if available: N/A

Description of changes:

Added docs to create a role which should have a trust relationship required to assume the role. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/contents/dev-docs/testing.md [hunk 1]
```diff
@@ -73,9 +73,58 @@ Finally, if you would like to do functional testing, where you would like to cre
 in your account using `KinD` cluster, please use `-r` and pass in `AWS ROLE ARN` to `scripts/kind-build-test.sh` script or
 `make kind-cluster-functional SERVICE=ecr ROLE_ARN=arn:aws:iam::12345678980:role/Admin-k8s`
 
-Note: For above functional test, `sts-assume-role.sh` under `scripts/lib` directory will 
-make `aws sts assume-role --role-session-arn $AWS_ROLE_ARN --role-session-name $TEMP_ROLE` API call to fetch `AWS_ACCESS_KEY_ID`, 
-`AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN` variable and inject them as secrets to the deployment. The duration of the session token is 900 seconds (15 minutes)
+Note: For above functional testing, `generate_temp_creds` function under `scripts/lib/aws.sh` script will
```
Comment (mhausenblas) [inline docs/contents/dev-docs/testing.md:76]: Please use the respective [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/), in this case `!!! note`.

File: docs/contents/dev-docs/testing.md [hunk 2]
```diff
@@ -73,9 +73,58 @@ Finally, if you would like to do functional testing, where you would like to cre
...
 `make kind-cluster-functional SERVICE=ecr ROLE_ARN=arn:aws:iam::12345678980:role/Admin-k8s`
 
-Note: For above functional test, `sts-assume-role.sh` under `scripts/lib` directory will 
-make `aws sts assume-role --role-session-arn $AWS_ROLE_ARN --role-session-name $TEMP_ROLE` API call to fetch `AWS_ACCESS_KEY_ID`, 
-`AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN` variable and inject them as secrets to the deployment. The duration of the session token is 900 seconds (15 minutes)
+Note: For above functional testing, `generate_temp_creds` function under `scripts/lib/aws.sh` script will 
+make `aws sts assume-role --role-session-arn $AWS_ROLE_ARN --role-session-name $TEMP_ROLE` API call to fetch `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN`. 
+The duration of the session token is 900 seconds (15 minutes). These variables will be added as Environment variables to the controller (deployment). 
+
+!!! The IAM user should be able to assume the role passed above else we can generate temporary credentials.
```
Comment (mhausenblas) [inline docs/contents/dev-docs/testing.md:80]: What admonition type is this? `note` or `tip` or `warning`. Also, did you preview this locally with `mkdocs serve`?

File: docs/contents/dev-docs/testing.md [hunk 3]
```diff
@@ -73,9 +73,58 @@ Finally, if you would like to do functional testing, where you would like to cre
...
+      "Version": "2012-10-17",
+      "Statement": {
+          "Effect": "Allow",
+          "Principal": { "AWS": "arn:aws:iam::12345678970:user/kubernetes" },
+          "Action": "sts:AssumeRole"
+      }
+    }
+    ```
+
+2. If you do not have a role, please run below commands to create a role, add the trust relationship to the role along with a sample policy arn
```
Comment (mhausenblas) [inline docs/contents/dev-docs/testing.md:108]: Consistency/style: drop stuff like "please" ;)

**Comment by nithu0115** [comment]
Thanks @mhausenblas for the review and pointers :) I forgot about mkdocs and was under the impression of markdown docs. I updated the PR and ran them locally.

**Comment by mhausenblas** [comment]
Thanks a lot @nithu0115 and that looks very good, content-wise. One last thing before we merge: can you share what mkdocs version you tested with, since we [upgraded to 5.x](https://squidfunk.github.io/mkdocs-material/upgrading/#upgrading-from-4x-to-5x) and I'm seeing some issue. For example, I have:

```
$ mkdocs  --version
mkdocs, version 1.1.2 from /usr/local/lib/python3.8/site-packages/mkdocs (Python 3.8)
```

**Comment by nithu0115** [comment]
> Thanks a lot @nithu0115 and that looks very good, content-wise. One last thing before we merge: can you share what mkdocs version you tested with, since we [upgraded to 5.x](https://squidfunk.github.io/mkdocs-material/upgrading/#upgrading-from-4x-to-5x) and I'm seeing some issue. For example, I have:
> 
> ```
> $ mkdocs  --version
> mkdocs, version 1.1.2 from /usr/local/lib/python3.8/site-packages/mkdocs (Python 3.8)
> ```
```
a483e7384446:aws-controllers-k8s nithmu$ sudo mkdocs --version
mkdocs, version 1.1.2 from /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/mkdocs (Python 3.8)
```

**Pull request merged**